fakebook/src/app/component/newsfeed/newsfeed.component.ts

```diff
@@ -12,22 +13,43 @@ import { PostService } from '../../service/post.service';
 export class NewsfeedComponent implements OnInit {
   posts: Post[] = [];
   user: User | null = null;
+  postId: number;
 
   constructor(
     private newsfeedService: NewsfeedService,
-    private postService: PostService
-  ) {}
+    private postService: PostService,
+    private route: ActivatedRoute
+  ) {
+    this.route.params.subscribe(params => {
+      console.log(params);
+      if(params['id'] != undefined){   
+        this.postId = +params['id'];
+          if(this.postId != undefined)
+            this.getPostsById();
+      }
+      else{
+        this.getPosts();
+      }
+    })
+  }
 
   ngOnInit(): void {
     this.getUser();
-    this.getPosts();
+    //this.getPosts();
   }
 
   getPosts(): void {
     this.newsfeedService
-      .getPosts()
+      .getPosts(this.postId)
+      .subscribe((gotPosts) => (this.posts = gotPosts));
+  }
+
+  getPostsById(): void {
```

Comment:
why a plural method name when it's going to be just one post?

Reply:
Renamed method.